web/src/connectors/task-annotator-connector/task-annotator-context.tsx

```diff
@@ -208,6 +238,7 @@ export const TaskAnnotatorContextProvider: React.FC<ProviderProps> = ({
 
     const [currentPage, setCurrentPage] = useState<number>(1);
 
+    const [annotationsChanges, setAnnotationsChanges] = useState<Record<number, Annotation[]>>({});
```

Comment:
In the scope of this PR we start to store all the changes related to annotations in the separate map. Before changes were just directly added to the total list of annotations. It's needed because we don't store all the items in the total list of annotations now when lazy loading mode is enabled for PDF (in lazy loading mode this list is constantly changed since we loading a new data during scroll and remove old cached data). Now, these changes are fully tracked separately.